engine/src/main/java/com/datatorrent/stram/engine/GenericNode.java

```diff
@@ -343,12 +389,15 @@ else if (!doCheckpoint) {
                  * we will receive tuples which are equal to the number of input streams.
                  */
                 activePort.remove();
-                buffers.remove();
+                if (isInputPortConnectedToDelayOperator(activePortEntry.getKey())) {
+                  break; // breaking out of the switch/case
+                }
 
+                buffers.remove();
```

Comment:
if ```buffers.remove()``` is before the ```if``` condition, the port will not be considered as active any more for this window, which may already have started since the other ports may have sent BEGIN_WINDOW before we get the RESET_WINDOW from the port connected to the delay operator.  In that case, the operator would get stuck.